src/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/Activity.cs

```diff
@@ -378,28 +381,43 @@ private string generateRootId()
                 // Here we make an ID to represent the Process/AppDomain.   Ideally we use process ID but 
                 // it is unclear if we have that ID handy.   Currently we use low bits of high freq tick 
                 // as a unique random number (which is not bad, but loses randomness for startup scenarios).  
-                Interlocked.CompareExchange(ref s_uniqPrefix, generateUniquePrefix(), null);
+                Interlocked.CompareExchange(ref s_uniqPrefix, generateInstancePrefix(), null);
             }
 
 #if DEBUG
-            string ret = s_uniqPrefix + OperationName + s_idDelimiter + Interlocked.Increment(ref s_currentRootId);
+            string ret = s_uniqPrefix + "-" + OperationName + "-" + Interlocked.Increment(ref s_currentRootId).ToString("x") + s_internalIdDelimiter;
 #else           // To keep things short, we drop the operation name 
-            string ret = s_uniqPrefix + s_idDelimiter + Interlocked.Increment(ref s_currentRootId);
+            string ret = s_uniqPrefix + "-" + Interlocked.Increment(ref s_currentRootId).ToString("x") + s_internalIdDelimiter;
 #endif
             return ret;
         }
 
+        private string _rootId;
+
         // Used to generate an ID 
-        private int _currentChildId;            // A unique number for all children of this activity.  
-        private static int s_currentRootId;      // A unique number inside the appdomain.
-        private static string s_uniqPrefix;
+        static Activity()
+        {
+            s_random = new Random();
+            
+            //Randomized on different process instances
+            s_currentRootId = s_random.Next();
+        }
 
-        private string _rootId;
-        private static char s_idDelimiter = '.';
+        private static readonly Random s_random;
+
+        // A unique number for all children of this activity.  
+        private int _currentChildId;
+
+        // A unique number inside the appdomain. 
+        private static long s_currentRootId;
+
+        private static string s_uniqPrefix;
+        private static char s_internalIdDelimiter = '.';
```

Comment:
Can we make all of these const (and drop the s_ and capitalize the variable (I believe this is our convention for constants, which is normal Pascal casing).   As it is they don't get inlined.